### PR TITLE
Consume latest version of ast-types

### LINKED ISSUE #463: React-docgen fails to parse components with optional chaining.
If a component contains optional chaining it will now fail to parse. This used to work.

I'm on version 4.1.1.

Now you get this error:

```
Error: did not recognize object of type "ChainExpression"
    at Object.getFieldNames (/proj/node_modules/ast-types/lib/types.js:661:19)
    at visitChildren (/proj/node_modules/ast-types/lib/path-visitor.js:186:36)
    at Visitor.PVp.visitWithoutReset (/proj/node_modules/ast-types/lib/path-visitor.js:168:20)
    at visitChildren (/proj/node_modules/ast-types/lib/path-visitor.js:205:25)
    at Visitor.PVp.visitWithoutReset (/proj/node_modules/ast-types/lib/path-visitor.js:168:20)
    at NodePath.each (/proj/node_modules/ast-types/lib/path.js:89:26)
    at visitChildren (/proj/node_modules/ast-types/lib/path-visitor.js:180:18)
    at Visitor.PVp.visitWithoutReset (/proj/node_modules/ast-types/lib/path-visitor.js:168:20)
    at visitChildren (/proj/node_modules/ast-types/lib/path-visitor.js:205:25)
    at Visitor.PVp.visitWithoutReset (/proj/node_modules/ast-types/lib/path-visitor.js:168:20)
```

I kind of suspect this has to do with one of the transitive dependencies more that react-docgen but I'm logging this here first.

How to recreate, run the following test against the below jsx file.

test.js
```js
const reactDocs = require('react-docgen');
const fs = require('fs');

const filePath = './file.jsx';

const source = fs.readFileSync(filePath, 'utf8');

const parsedProps = reactDocs.parse(source);

console.log('parsed', parsedProps);

```

file.jsx
```jsx
import React from 'react';

const ApplicationBase = () => {
  const thing = {
    stuff: 'stuff',
  };

  // Parse works when the next line is commented out.
  const string = thing?.stuff;
  // Works when un-commented
  // const string = 'stuff';

  return (
    <div>
      {string}
    </div>
  );
};

export default ApplicationBase;

```

Also an issue template would help me remember all the things you'd like to see on an issue 😄

### PR DESCRIPTION
**Pull request by mjhenkes**
Closes #463

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1180,10 +1180,12 @@ assign-symbols@^1.0.0:
...
-ast-types@^0.13.2:
-  version "0.13.2"
-  resolved "https://registry.yarnpkg.com/ast-types/-/ast-types-0.13.2.tgz#df39b677a911a83f3a049644fb74fdded23cea48"
-  integrity sha512-uWMHxJxtfj/1oZClOxDEV1sQ1HCDkA4MG8Gr69KKeBjEVH0R84WlejZ0y2DcwyBlpAEMltmVYkVgqfLFb2oyiA==
+ast-types@^0.14.2:
+  version "0.14.2"
+  resolved "https://registry.yarnpkg.com/ast-types/-/ast-types-0.14.2.tgz#600b882df8583e3cd4f2df5fa20fa83759d4bdfd"
+  integrity sha512-O0yuUDnZeQDL+ncNGlJ78BiO4jnYI3bvMsD5prT0/nsgijG/LpNBIr63gTjVTNsiGkgQhiyCShTgxt8oXOrklA==
+  dependencies:
+    tslib "^2.0.1"
```
Comment (mjhenkes) [inline yarn.lock:1188]: I suspect this should have been a dev dependency. Asked the question here: https://github.com/benjamn/ast-types/issues/415

**Comment by ethancrook99** [comment]
Thanks for updating this!

File: yarn.lock (hunk 1 above)
Comment (mjhenkes) [reply yarn.lock:1188]: Confirmed that tslib was correctly added as a dependency.

**Comment by danez** [comment]
Thanks

**Pull request merged**